### PR TITLE
Change randomized prefix to contain only hex characters

### PR DESCRIPTION
**Pull request by arhimondr**
Some storage systems prefer the prefix to be hexadecimal characters

No release notes entries required

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java [hunk 1]
```diff
@@ -67,7 +67,7 @@
         implements Exchange
 {
     private static final Pattern PARTITION_FILE_NAME_PATTERN = Pattern.compile("(\\d+)_(\\d+)\\.data");
-    private static final char[] RANDOMIZED_PREFIX_ALPHABET = "abcdefghijklmnopqrstuvwzyz0123456789".toCharArray();
+    private static final char[] RANDOMIZED_PREFIX_ALPHABET = "abcdef0123456789".toCharArray();
```
Comment (electrum) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java:70]: Please add a comment explaining why this exact alphabet is important.

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java [hunk 2]
```diff
@@ -67,7 +67,7 @@
...
     private static final int RANDOMIZED_PREFIX_LENGTH = 6;
```
Comment (electrum) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java:71]: Do we need to increase this now? This takes us from `2176782336` combinations to only `16777216`, or 0.0077% of the previous.

**Comment by electrum** [comment]
Since we're enforcing hex, how about simplifying the generation function:
```java
private static final int RANDOMIZED_PREFIX_BYTES = 3;

private static String generateRandomizedPrefix()
{
    byte[] bytes = new byte[RANDOMIZED_PREFIX_BYTES];
    ThreadLocalRandom.current().nextBytes(bytes);
    return BaseEncoding.base16().encode(bytes);
}

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java (hunk 1 above)
Comment (arhimondr) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java:70]: Discussed offline

**Comment by arhimondr** [comment]
@electrum Having the custom function avoids unnecessary allocations and reduces uncertainty whether a lower case or upper case characters are used

**Review by electrum — APPROVED** [review]

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java (hunk 2 above)
Comment (findepi) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java]: Should these assumptions be persisted as a code comment, documenting why 6 is still sufficient prefix length?

**Pull request merged**